pkgs/development/tools/comby/default.nix

```diff
@@ -0,0 +1,66 @@
+{ ocamlPackages, fetchFromGitHub, lib, zlib, pkg-config }:
+let
+  mkCombyPackage = { pname, extraBuildInputs ? [ ], extraNativeInputs ? [ ] }:
+    ocamlPackages.buildDunePackage rec {
+      inherit pname;
+      version = "1.5.1";
+      useDune2 = true;
+      minimumOcamlVersion = "4.08.1";
+
+      src = fetchFromGitHub {
+        owner = "comby-tools";
+        repo = "comby";
+        rev = version;
+        sha256 = "1ipfrr6n1jyyryhm9zpn8wwgzfac1zgbjdjzrm00qcwc17r8x2hf";
+      };
+
+      nativeBuildInputs = [
+        ocamlPackages.ppx_deriving
+        ocamlPackages.ppx_deriving_yojson
+        ocamlPackages.ppx_sexp_conv
+        ocamlPackages.ppx_sexp_message
+      ] ++ extraNativeInputs;
+
+      buildInputs = [
+        ocamlPackages.core
+        ocamlPackages.ocaml_pcre
+        ocamlPackages.mparser
+        ocamlPackages.mparser-pcre
+      ] ++ extraBuildInputs;
+
+      meta = {
+        description = "Tool for searching changing code structure";
+        license = lib.licenses.asl20;
+        homepage = "https://comby.dev";
+      };
+    };
+
+  combyKernel = mkCombyPackage {
+    pname = "comby-kernel";
+    extraBuildInputs = [ ocamlPackages.re ocamlPackages.angstrom ];
+    extraNativeInputs = [ ocamlPackages.ppx_here ];
+  };
+in mkCombyPackage {
+  pname = "comby";
+
+  extraBuildInputs = [
+    zlib
+    ocamlPackages.shell # This input must appear before `parany` or any other input that propagates `ocamlnet`
+    ocamlPackages.lwt
+    ocamlPackages.patience_diff
+    ocamlPackages.toml
+    ocamlPackages.cohttp-lwt-unix
+    ocamlPackages.opium
+    ocamlPackages.textutils
+    ocamlPackages.jst-config
+    ocamlPackages.parany
+    combyKernel
+  ];
```

Comment:
opam file also lists:

* conduit-lwt-unix
* hack_parallel
* lwt_react
* tls

Maybe there are optional dependencies as well? It lists the system dependencies gmp, libev as well…

Reply:
I’ve added the dependencies to the inputs. Maybe some of them are runtime dependencies and the program would crash under certain circumstances if they’re not included?